### ISSUE TITLE
Expand the list of aspects that are required to make good maps

### ISSUE THREAD

**Issue by tsamsonov** [r-tmap/tmap-book#20]
Hi @mtennekes and @Nowosad 

Thank you for work on this book and also for making the great `tmap` package more powerful and accessible for R users.

As a geographer with specialization in cartography I see two important aspects that are missing in the introduction:

1. _Knowledge of geography_. Many errors in maps are made by non-professionals just because they do not know how the world works. Since the computed maps are generated from data, it is the data, processing and visualization blunders that induce geographic errors in maps. Imagine a combination of settlement and hydrographic data taken from different sources. If the lines are simplified, it may appear that settlement points are located on the wrong side of rivers. I distinguish it from the domain knowledge, which is specific, or thematic: soils, oceans, transportation, etc. It is more the general knowledge of the geographic space. It is undoubtedly, that making maps with R is needed for many non-professionals, and it is unreasonable to expect a good knowledge of geography from any user. But at the same time it is also a good idea to encourage users to check the geographic correctness of their maps using a well-trusted reference sources, such as the Times atlas of the world. And to use the well-trusted datasets, too.

2. _Knowledge of established cartographic practices_. These are not equal to data visualization knowledge. Surely, the latter are essential to use the visual language effectively while making maps. However, a good cartography requires a wider visualization background. For example, it is impossible to make a correct world elevation map from DEM without extracting the land part and coloring it separately from oceanic. Otherwise, the areas with negative elevation, such as the Caspian depression, will be coloured with 'wrong' blue colors (they should be dark green, if the conventional elevation scale is used). Another exclusively cartographic visualization issue is about map projections. The user should have at least the general knowledge of them, and how the projection distorts the representation.

Surely, the list of examples can be expanded. But the R user must be bewared that just knowing how to visualize a spatial data frame with proper visual variable is not enough to make a a good geographic map. That's why cartography established as a technical and scientific discipline.

All the best, Tim

**Comment by mtennekes** [comment]
Thanks for your feedback! Interesting food for discussion:-)

I mostly agree with 1, but at the same time I also think it that "domain" and "geographic space" are two sides of the same coin. For example, I don't know if there are any domain experts of transportation in Italy who don't have knowledge of the geographic space of Italy. The same for the other domains you mention.

To be honest I think that the discipline of cartography is contained in the intersection of (general) data visualization and spatial data science. I believe that someone with those specializations (with domain/geography knowledge) should be able to make good maps.

Side note: I'm always a bit skeptic when it comes to 'well-trusted' and 'established'. Such qualifications are good, but don't necessarily imply that they always should be followed (as many established theories/practices turned out to be false or outdated).

**Comment by mtennekes** [comment]
Forgot to mention that I absolutely agree with:

> Surely, the list of examples can be expanded. But the R user must be bewared that knowing how to visualize a spatial data frame with proper visual variable does not prevent them from geographic errors in their maps.

We are not settled on the outline yet. The CRS part is currently covered in Chapter 2 (2.4), but ideally it should be somewhere else, because it is a broader subject (broader than Spatial data in R). What do you think, @Nowosad ? @tsamsonov Have you read this section 2.4 yet? Feedback from someone with your background is highly appreciated.

We plan to write a separate chapter about making good maps (in the current outline chapter 15). This will cover many general data visualization aspects as well as these established cartographic practices as you call them. I've seen way to many wrong maps circulating on newspapers and social media. For instance, choropleths of absolute values (instead of ratios or densities).
This chapter should cover all aspects needed to make good maps, including the two you mention (irrespective of how you classify and name them).

**Comment by tsamsonov** [comment]
Hi @mtennekes and thanks for your thoughts!

> I mostly agree with 1, but at the same time I also think it that "domain" and "geographic space" are two sides of the same coin. For example, I don't know if there are any domain experts of transportation in Italy who don't have knowledge of the geographic space of Italy. The same for the other domains you mention.

While speaking about domain, I think that there are a lot of domains in which the spatial context is not primary, and thus is not learned extensively (or maybe is not learned at all) at university courses. But some fraction of those specialists sooner or later come to the need to make a map. Take economics or healthcare, for example. Or, even better, a journalism. These specialists can be very confident in their domain. And some of them need to make a map of transport accessibility, lung cancer rates or military operations. Do we expect that they are experts in geography, too? I'd rather say no. That is why I think that "domain" knowledge and understanding of "geographic space" are very different things. 

> To be honest I think that the discipline of cartography is contained in the intersection of (general) data visualization and spatial data science. I believe that someone with those specializations (with domain/geography knowledge) should be able to make good maps.

It is very interesting topic! Cartography is a very old discipline, and it was established far long before dataviz and data science domains emerged. I think that you are more or less correct in suggesting its current position in the field of sciences, but it is still somewhat approximate. 

> Side note: I'm always a bit skeptic when it comes to 'well-trusted' and 'established'. Such qualifications are good, but don't necessarily imply that they always should be followed (as many established theories/practices turned out to be false or outdated).

Yes, a true researcher is always sceptic about any theory ;) But I am talking more about geographic facts, not theories. Knowing that the Washington D.C. is on the left side of the Potomak river is a fact. Also, a negative value of elevation does not mean that the area is underwater. There are myriads of such facts. And since knowing all of them is impossible, it is good to check a map against a trusted source of geographic information. Or maintain a carefully checked spatial data source such as Natural Earth.

**Comment by tsamsonov** [comment]
> We are not settled on the outline yet. The CRS part is currently covered in Chapter 2 (2.4), but ideally it should be somewhere else, because it is a broader subject (broader than Spatial data in R). What do you think, @Nowosad ? @tsamsonov Have you read this section 2.4 yet? Feedback from someone with your background is highly appreciated.

From the first glance I think that projections deserve a separate chapter, since projecting is often a stage of data visualization. Maybe it would be reasonable to cover the CRS representation and handling (WKT, PROJ and all that technical stuff) inside the Spatial data chapter, while explaining the projections in a separate chapter later. You can do this in more detail than can be typically found in spatial data science books, since projections a crucial for mapping.

I am interested to read all topics in this book and give my feedback in any case when I feel it can be valuable, or something is missing.  

> We plan to write a separate chapter about making good maps (in the current outline chapter 15). This will cover many general data visualization aspects as well as these established cartographic practices as you call them. I've seen way to many wrong maps circulating on newspapers and social media. For instance, choropleths of absolute values (instead of ratios or densities). This chapter should cover all aspects needed to make good maps, including the two you mention (irrespective of how you classify and name them).

Since those 'best practices' will inevitably be scattered across all book chapters, I think it is a good idea to write a separate chapter which summarizes them 👍🏻

I must note that I usually discourage my colleagues from using the subjective terms, such as "good" or "elegant",  in describing their own scientific work. It obliges you to a lot of things. This can be effective for marketing purposes (if you want to sell more books), but honestly must be substantiated by extensive feedback from recognized dataviz/cartography experts which reviewed your maps. There is always a trap of being in subjective illusion that your maps are very nice, while someone will find a lot of downsides in it. But I agree that there is a need for specific term that characterizes a map that is made with all those best practices and dataviz/spatial knowledge in mind. And it is not easy to formulate.

**Comment by mtennekes** [comment]
Thanks for sharing your reflections; I agree with all of them. We'll let you know when chapters are ready for a review. Probably early next year.

I absolutely agree that subjective terms such as good and elegant should be avoided. Any ideas for a better title @Nowosad ?

**Comment by Nowosad** [comment]
I agree with both of you that we should avoid subjective terms in the text of the book. At the same time, I think it is fine for a title to use them. The role of the title is to attract potential readers and not to tell the story. I think the current title is fine, but let me know if you have any better alternatives or like some of [other ideas](https://github.com/r-tmap/tmap-book/blob/master/meta/book-title.Rmd) more?
